delphi/translators/for2py/rectify.py

```diff
@@ -177,60 +413,136 @@ def handle_tag_header(self, root, parElem):
+                            cur_elem.attrib['code'] = str(rand_code)
+
+                        if parent.tag == "loop":
+                            if (
+                                child.tag == "if"
+                                or child.tag == "statement"
+                                and "conditional-goto-stmt" in child.attrib
+                            ):
+                                self.goto_under_loop = True
+                    else:
+                        # Reconstruction of statements
+                        if not self.goto_under_loop:
+                            if (
+                                "parent" in current.attrib 
+                                and current.attrib['parent'] == "program"
+                            ):
+                                # Remove statements that is marked to be removed (2nd traverse)
+                                if (
+                                    "goto-remove" in child.attrib 
+                                    or "goto-move" in child.attrib
+                                ):
+                                    current.remove(cur_elem)
+
+                                    if self.reconstruct_after_case_now and not self.reconstruction_for_after_done:
+                                        self.reconstruct_goto_after_label(new_parent, traverse, self.statements_to_reconstruct_after)
+                                        if self.label_lbl_for_before:
+                                            self.continue_elimination = True
+                                    if self.reconstruct_before_case_now and not self.reconstruction_for_before_done:
+                                        reconstruct_target = self.statements_to_reconstruct_before
+                                        self.reconstruct_goto_before_label(new_parent, traverse, reconstruct_target)
+                                        if self.label_lbl_for_after :
+                                            self.continue_elimination = True
+                                    if not self.label_lbl_for_before and not self.label_lbl_for_after:
+                                        self.continue_elimination = False
+                        else:
+                            if (
+                                "parent" in current.attrib
+                                and current.attrib['parent'] == "loop"
+                            ):
+                                # Remove statements that is marked to be removed (2nd traverse)
+                                if (
+                                    "goto-remove" in child.attrib 
+                                    or "goto-move" in child.attrib
+                                ):
+                                    current.remove(cur_elem)
+
+                                    if self.reconstruct_after_case_now and not self.reconstruction_for_after_done:
```

Comment:
Lines 518-528 seem very similar to lines 495-505.  Is it possible to avoid the code duplication?